### DISCUSSION TITLE
reComputer R series will release versions with more IO and bus interfaces in the future. Please send your wish list to us.

### DISCUSSION THREAD

**Discussion by Ainsfei** [Seeed-Studio/OSHW-RPi-Series#7] [Open for vote]
Hi everyone,

Need more IO and interface? You name it we make it! If you can inform us about your usage scenarios and interface requirements, there will be a higher chance of adoption.

**Comment by jacobertel** [comment]
It would be nice to have a WiFi Halo module for the reComputer R1000. This wouldn't even require a hardware update, as I'm assuming it would just slot in as a mini-PCIe module. Thanks!

**Comment by robot-ranger** [comment]
First - The major IO feature we are missing for our use is more ethernet ports. We connect several ethernet devices (ex multiple machines, multiple Modbus IO blocks). Lets be real - all communication is moving toward ethernet-based protocols and away from DIO and serial connections. Existing solutions on the market, including R1000, only have 2 ports which requires us to also supply an ethernet switch. This means we also have to power the ethernet switch. IMO; the auxiliary switch could be eliminated by integrating more ports (or even a basic unmanaged ethernet switch; we could implement a dhcp if needed), into the R1000. [LIKE THIS](https://www.aliexpress.us/item/3256807167080278.html?spm=a2g0o.productlist.main.73.3ddb5e8e3eanOQ&algo_pvid=c84a9c9c-3cb7-49b9-ad4c-e05dc3ba0d7d&algo_exp_id=c84a9c9c-3cb7-49b9-ad4c-e05dc3ba0d7d-36&pdp_npi=4%40dis%21USD%2134.99%2134.99%21%21%2134.99%2134.99%21%402103010b17279409549895650e9d36%2112000040390805958%21sea%21US%210%21ABX&curPageLogUid=jyeRVMdfTZ0r&utparam-url=scene%3Asearch%7Cquery_from%3A) We need 4-5 external ethernet connections.

Second - personally; i really dont like control cabinets. I wish the R1000 was more 'hardened' for use outside the control cabinet. 
A tight case and "industrial" style connectors would be perfect; We dont need IP67 certification or waterproof, but hdmi ports, rj45 ports, etc are not quite enough either. Many manufacturing facilities that we work in are dusty and dirty. Over time, connectors like USB/rj45/hdmi, etc get impacted by the dusty dirty environment. Its not an issue if there is a plug inserted, so maybe the USB and HDMI ports could have a screw-fastened hatch designed into the case?

I would really appreciate a version with typical 'fieldbus' style m12 connectors:

- M12 L-Coded for Power
- 4-5x M12 (X-Coded if gigabit; D-Coded if 10/100M) ethernet
- DB9 for COM/serial
- M12 A-coded 8-pin for DIO (would be nice if pin-compatible with common IO blocks [LIKE THIS](https://www.ifm.com/us/en/product/EBC050) to allow for use of off-the-shelf cables)
- Internal SD and SIM card are acceptable; they dont get serviced often

All of these connectors provide highly secure, threaded, connections and use overmolded, off-the-shelf, cables. 

No one else on the market makes a compact ARM-based industrial PC with "industrial" connectors at a reasonable price. They do exist, but there is a giant leap in price from control-cabinet raspberry pi devices to train/railcar industry computers that do have the industrial connectors I listed above, but also come with full IP67/waterproof ratings. And when I say giant leap in price - I mean like easily >$5kUSD for essentially a raspberryPi in a waterproof box.

RevPi, OnLogic, etc only make DIN-rail Pi-based devices, but theyre for inside the control cabinet. NanoPi's R5S is close, but again uses RJ45 etc.
